Wikipedia/Code/WMFWelcomeContainerViewController.swift

```diff
@@ -42,59 +45,64 @@ class WMFWelcomeContainerViewController: UIViewController {
         switch welcomePageType {
         case .intro:
             return false
+        case .exploration:
+            return true
         case .languages:
             return true
         case .analytics:
             return true
         }
     }
 
-    fileprivate func hideAndCollapseTopContainerViewIfDeviceIsiPhone4s() {
-        if view.frame.size.height == 480 {
-            topContainerView.alpha = 0
-            topContainerViewHeightConstraint.constant = 0
+    override func updateViewConstraints() {
+        super.updateViewConstraints()
+        if needsDeviceAdjustments {
+            if isiPhone5() {
+                reduceTopAnimationsSizes(reduction: 50)
+            }
+            useBottomAlignmentIfPhone()
+            if isiPhone4s() {
+                hideAndCollapseTopContainerView()
+            }
+            needsDeviceAdjustments = false
         }
     }
     
+    fileprivate func isiPhone5() -> Bool {
+        return view.frame.size.height == 568
```

Comment:
Maybe this should be `> 400 && < 568` instead of looking for an exact match? It's possible new short devices are introduced that aren't exactly 568 points tall

Reply:
Fixed!

Reply:
Not using UIScreen.main.bounds.size.height for these ended up being the real reason why my code reducing the size of the top animation views didn't work without calling view.superview?.layoutIfNeeded() here: https://github.com/wikimedia/wikipedia-ios/pull/1803/files#diff-7b043c3999562fd7b1ebad3e7dcc4244L31